api/include/opentelemetry/metrics/meter.h

```diff
@@ -91,6 +94,25 @@ class Meter
       nostd::string_view description = "",
       nostd::string_view unit        = "") noexcept = 0;
 
+  /**
+   * Creates a Gauge with the passed characteristics and returns a unique_ptr to that Counter.
+   *
+   * @param name the name of the new Gauge.
+   * @param description a brief description of what the Gauge is used for.
+   * @param unit the unit of metric values following https://unitsofmeasure.org/ucum.html.
+   * @return a unique pointer to the created Gauge.
+   */
+
+  virtual nostd::unique_ptr<Gauge<int64_t>> CreateInt64Gauge(
+      nostd::string_view name,
+      nostd::string_view description = "",
+      nostd::string_view unit        = "") noexcept = 0;
+
+  virtual nostd::unique_ptr<Gauge<double>> CreateDoubleGauge(
+      nostd::string_view name,
+      nostd::string_view description = "",
+      nostd::string_view unit        = "") noexcept = 0;
```

Comment:
These are new virtual methods in the API, which breaks the ABI.

Please use:

```
#if OPENTELEMETRY_ABI_VERSION_NO >= 2
#endif
```

to define them in ABI version 2 only, as ABI version 1 can not change.

Reply:
Thanks for the quick feedback. 

I updated the code accordingly.

BTW I believe it still won't pass the format test since I could not find clang-format-10 for the debian distribution I'm using.

Reply:
There is a helper docker image to format:

```
docker run --rm --privileged=true --volume ${PWD}:/otel otel/cpp_format_tools
```

Run this from the PR top level directory.

Reply:
Thanks. As the last issue, a test case in Bazel valgrind CI (otlp_file_exporter_test) is failing. I'm not sure how it is related to this PR since it seems to be trace related. Could someone check and give me some pointers on how to fix maybe?

Reply:
This test failure is a known existing issue, please ignore it.

Reply:
Formatting should be fixed now.